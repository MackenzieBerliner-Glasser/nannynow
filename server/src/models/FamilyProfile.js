import mongoose from 'mongoose';
import { getStates } from './helpers/listHelpers';

const familyProfileSchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Contact name is required.']
  },
  streetAddress1: {
    type: String,
    required: [true, 'Street address is required.']
  },
  streetAddress2: String,
  city: {
    type: String,
    required: [true, 'City is required.']
  },
  state: {
    type: String,
    required: [true, 'State is required.'],
    enum: getStates()
  },
  zip: {
    type: Number,
    min: 501,
    max: 99950,
    required: [true, 'ZIP code is required.']
  },
  phone: {
    type: Number,
    required: [true, 'State is required.']
  },
  email: {
    type: String,
    required: [true, 'Email is required']
  },
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  agency: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Agency',
    required: [true, 'Agency is required']
  },
  description: {
    type: String,
    required: [true, 'Description is required']
  },
  createdDate: {
    type: Date,
    default: Date.now()
  },
  numOfChildren: {
    type: Number,
    required: [true, 'Number of children is required.']
  },
  birthdays: [
    {
      type: String,
      required: [true, 'children\'s birthdays are required.']
    }
  ]
});

export default mongoose.model('FamilyProfile', familyProfileSchema);
