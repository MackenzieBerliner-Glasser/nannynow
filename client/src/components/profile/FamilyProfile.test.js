import React from 'react';
import { shallow } from 'enzyme';
import FamilyProfile from './FamilyProfile';
import { fakeUser } from '../../reducers/fixtures/fakeUser';


describe('Profile component', () => {
  it('renders a component', () => {
    const createProfile = jest.fn();
    const updateProfile = jest.fn();
    const wrapper = shallow(<FamilyProfile createProfile={createProfile} updateProfile={updateProfile} session={fakeUser} />);
    expect(wrapper).toMatchSnapshot();
  });



});
