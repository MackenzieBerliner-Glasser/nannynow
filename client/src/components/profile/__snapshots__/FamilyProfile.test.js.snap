// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders a component 1`] = `
<Fragment>
  <h1
    id="profileHeader"
  >
    Create Your Profile
  </h1>
  <div
    id="profileBody"
  >
    <form
      id="profileForm"
      onSubmit={[Function]}
    >
      <label
        htmlFor="name"
      >
        Full Name:
      </label>
      <input
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="streetAddress1"
      >
        Street Address:
      </label>
      <input
        name="streetAddress1"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <input
        name="streetAddress2"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="city"
      >
        City
      </label>
      <input
        name="city"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="state"
      >
        State
      </label>
      <select
        name="state"
        onChange={[Function]}
        required={true}
        value=""
      >
        <option
          value=""
        >
          --
        </option>
        <option
          key="AL"
          value="AL"
        >
          AL
        </option>
        <option
          key="AK"
          value="AK"
        >
          AK
        </option>
        <option
          key="AS"
          value="AS"
        >
          AS
        </option>
        <option
          key="AZ"
          value="AZ"
        >
          AZ
        </option>
        <option
          key="AR"
          value="AR"
        >
          AR
        </option>
        <option
          key="CA"
          value="CA"
        >
          CA
        </option>
        <option
          key="CO"
          value="CO"
        >
          CO
        </option>
        <option
          key="CT"
          value="CT"
        >
          CT
        </option>
        <option
          key="DE"
          value="DE"
        >
          DE
        </option>
        <option
          key="DC"
          value="DC"
        >
          DC
        </option>
        <option
          key="FM"
          value="FM"
        >
          FM
        </option>
        <option
          key="FL"
          value="FL"
        >
          FL
        </option>
        <option
          key="GA"
          value="GA"
        >
          GA
        </option>
        <option
          key="GU"
          value="GU"
        >
          GU
        </option>
        <option
          key="HI"
          value="HI"
        >
          HI
        </option>
        <option
          key="ID"
          value="ID"
        >
          ID
        </option>
        <option
          key="IL"
          value="IL"
        >
          IL
        </option>
        <option
          key="IN"
          value="IN"
        >
          IN
        </option>
        <option
          key="IA"
          value="IA"
        >
          IA
        </option>
        <option
          key="KS"
          value="KS"
        >
          KS
        </option>
        <option
          key="KY"
          value="KY"
        >
          KY
        </option>
        <option
          key="LA"
          value="LA"
        >
          LA
        </option>
        <option
          key="ME"
          value="ME"
        >
          ME
        </option>
        <option
          key="MH"
          value="MH"
        >
          MH
        </option>
        <option
          key="MD"
          value="MD"
        >
          MD
        </option>
        <option
          key="MA"
          value="MA"
        >
          MA
        </option>
        <option
          key="MI"
          value="MI"
        >
          MI
        </option>
        <option
          key="MN"
          value="MN"
        >
          MN
        </option>
        <option
          key="MS"
          value="MS"
        >
          MS
        </option>
        <option
          key="MO"
          value="MO"
        >
          MO
        </option>
        <option
          key="MT"
          value="MT"
        >
          MT
        </option>
        <option
          key="NE"
          value="NE"
        >
          NE
        </option>
        <option
          key="NV"
          value="NV"
        >
          NV
        </option>
        <option
          key="NH"
          value="NH"
        >
          NH
        </option>
        <option
          key="NJ"
          value="NJ"
        >
          NJ
        </option>
        <option
          key="NM"
          value="NM"
        >
          NM
        </option>
        <option
          key="NY"
          value="NY"
        >
          NY
        </option>
        <option
          key="NC"
          value="NC"
        >
          NC
        </option>
        <option
          key="ND"
          value="ND"
        >
          ND
        </option>
        <option
          key="MP"
          value="MP"
        >
          MP
        </option>
        <option
          key="OH"
          value="OH"
        >
          OH
        </option>
        <option
          key="OK"
          value="OK"
        >
          OK
        </option>
        <option
          key="OR"
          value="OR"
        >
          OR
        </option>
        <option
          key="PW"
          value="PW"
        >
          PW
        </option>
        <option
          key="PA"
          value="PA"
        >
          PA
        </option>
        <option
          key="PR"
          value="PR"
        >
          PR
        </option>
        <option
          key="RI"
          value="RI"
        >
          RI
        </option>
        <option
          key="SC"
          value="SC"
        >
          SC
        </option>
        <option
          key="SD"
          value="SD"
        >
          SD
        </option>
        <option
          key="TN"
          value="TN"
        >
          TN
        </option>
        <option
          key="TX"
          value="TX"
        >
          TX
        </option>
        <option
          key="UT"
          value="UT"
        >
          UT
        </option>
        <option
          key="VT"
          value="VT"
        >
          VT
        </option>
        <option
          key="VI"
          value="VI"
        >
          VI
        </option>
        <option
          key="VA"
          value="VA"
        >
          VA
        </option>
        <option
          key="WA"
          value="WA"
        >
          WA
        </option>
        <option
          key="WV"
          value="WV"
        >
          WV
        </option>
        <option
          key="WI"
          value="WI"
        >
          WI
        </option>
        <option
          key="WY"
          value="WY"
        >
          WY
        </option>
      </select>
      <label
        htmlFor="zip"
      >
        Zip
      </label>
      <input
        name="zip"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="phone"
      >
        Phone Number
      </label>
      <input
        name="phone"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="birthday"
      >
        How Old Are Your Children?
      </label>
      <input
        min="2002-12-12"
        name="birthday"
        onChange={[Function]}
        type="date"
        value=""
      />
      <button
        id="addChildButton"
        onClick={[Function]}
        type="button"
      >
        Add Child
      </button>
      <label
        htmlFor="description"
      >
        Tell Us About Your Family. Any Special Needs?:
      </label>
      <input
        name="description"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <button
        id="submitButton"
      >
        Submit Profile
      </button>
    </form>
    <div
      id="profileView"
    >
      <div
        id="doubleBorder1"
      >
        <div
          id="doubleBorder2"
        >
          <h1 />
          <h3>
            Street Address:
          </h3>
          <h3>
            Address (continued): 
          </h3>
          <h3>
            City: 
          </h3>
          <h3>
            State: 
          </h3>
          <h3>
            Zip Code: 
          </h3>
          <h3>
            Phone Number: 
          </h3>
          <h3>
            Number of Children: 
            0
          </h3>
          <h3>
            Child Birthdays: 
          </h3>
          <ul />
          <h3>
            Details: 
          </h3>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
